x-pack/plugins/security/public/telemetry/index.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { HttpStart } from 'src/core/public';
+
+import type { AuthenticatedUser } from '../../common';
+
+interface SecurityTelemetryStartParams {
+  http: HttpStart;
+  getCurrentUser: () => Promise<AuthenticatedUser>;
+}
+
+interface SecurityTelemetryAuthType {
+  username_hash: string;
+  timestamp: number;
+  auth_type: string;
+}
+
+export class SecurityTelemetryService {
+  public static KeyAuthType = 'kibana-user-auth-type';
+
+  public async start({ http, getCurrentUser }: SecurityTelemetryStartParams) {
+    // wait for the user to be authenticated before doing telemetry
+    getCurrentUser().then(() => this.postAuthTypeTelemetry(http));
```

Comment:
:+1: this seems to work well!

In the current implementation, Kibana makes this API call even if Security is not enabled (in Elasticsearch).
In that case, we could avoid making the call altogether.

The SecurityNavControlService does something similar by checking the security license:
https://github.com/elastic/kibana/blob/e5b279f318eac64660a5f4b157e891339f05ea61/x-pack/plugins/security/public/nav_control/nav_control_service.tsx#L64-L74

Only for this, we'd want to check the `allowRbac` field instead.

Something like this, perhaps?

```suggestion
    this.securityFeaturesSubscription = this.securityLicense.features$.subscribe(
      ({ allowRbac }) => {
        if (allowRbac) {
          getCurrentUser().then(() => this.postAuthTypeTelemetry(http));
        }
      }
    );
```